Refactor Journal.js into smaller submodules

jniles opened the pull request:
This PR will close #47.  However, there are still some items left to do.

 - [x] refactor sale route
 - [x] refactor sale discard route
 - [x] refactor cash payment route
 - [x] refactor cash refund route (used to be cash/discard)
 - [x] refactor primary cash route
 - [x] refactor primary cash income route
 - [x] refactor primary cash expense route
 - [x] refactor primary cash refund route
 - [x] refactor payroll route
 - [x] refactor salary payment route
 - [x] refactor credit note route
 - [x] refactor consumption loss route
 - [x] refactor donation route
 - [x] refactor purchase route
 - [x] refactor employee invoice route
 - [x] refactor employee payment route
 - [x] refactor transfer route
 - [x] refactor convention payment route
 - [x] refactor promise routes
 - [x] refactor advance route
 - [x] refactor confirm (?) route
 - [x] refactor fiscal year result (?) route
 - [x] refactor integration route
 - [x] refactor tax payment route
 - [x] refactor cotisation payment route
 - [x] reversing stock

### EDIT

All routes have been migrated.  Testing in progress...

DedrickEnc commented on server/controllers/journal/cash.js:
```diff
@@ -0,0 +1,411 @@
+var q = require('q'),
```
You put cash and cash discard logic in the same controller but why not for sale and sale discard? and also caution should be in the cash controller like cash and cash discard according to your logic.

jniles replied:
That's fine - I didn't have time to even touch sale discard.  You will notice that sale discard is not checked on the list.

jniles commented:
I got this one now.

jniles commented:
@sfount, @dedrickenc - The PR can be merged anytime.  I've tested as much as I know how to test.

There are two bugs unrelated to this PR that crash `#/group_invoice` and `#/support`.  I believe it was because of #653.  There is some pretty broken code in those modules, so I recommend addressing the issues with them in a PR independent of this one.

Once this lands, I'll be able to merge #670 and #690 .  Let me know if you find bugs!

sfount commented:
This is really great work Jon. It will help a **huge** amount with maintainability and future code updates. I will pull this request to make sure it works with my build this evening. I'm happy for this to be merged whenever.
